src/esp/gfx/PbrDrawable.cpp

```diff
@@ -129,42 +139,48 @@ void PbrDrawable::draw(const Mn::Matrix4& transformationMatrix,
   if (normalDet < 0) {
     Mn::GL::Renderer::setFrontFace(Mn::GL::Renderer::FrontFace::ClockWise);
   }
+  const auto& tmpMaterialData =
+      materialData_->as<Mn::Trade::PbrMetallicRoughnessMaterialData>();
 
   (*shader_)
       // e.g., semantic mesh has its own per vertex annotation, which has
       // been uploaded to GPU so simply pass 0 to the uniform "objectId" in
       // the fragment shader
-      .setObjectId(
-          static_cast<RenderCamera&>(camera).useDrawableIds()
-              ? drawableId_
-              : (materialData_->perVertexObjectId ? 0 : node_.getSemanticId()))
+      .setObjectId(static_cast<RenderCamera&>(camera).useDrawableIds()
+                       ? drawableId_
+                       : (materialData_->attribute<bool>("hasPerVertexObjectId")
+                              ? 0
+                              : node_.getSemanticId()))
       .setProjectionMatrix(camera.projectionMatrix())
       .setViewMatrix(camera.cameraMatrix())
       .setModelMatrix(modelMatrix)  // NOT modelview matrix!
       .setNormalMatrix(normalMatrix)
       .setCameraWorldPosition(
           camera.object().absoluteTransformationMatrix().translation())
-      .setBaseColor(materialData_->baseColor)
-      .setRoughness(materialData_->roughness)
-      .setMetallic(materialData_->metallic)
-      .setEmissiveColor(materialData_->emissiveColor);
+      .setBaseColor(tmpMaterialData.baseColor())
+      .setRoughness(tmpMaterialData.roughness())
+      .setMetallic(tmpMaterialData.metalness())
+      .setEmissiveColor(tmpMaterialData.emissiveColor());
 
   // TODO:
   // IN PbrShader class, we set the resonable defaults for the
   // PbrShader::PbrEquationScales. Here we need a smart way to reset it
   // just in case user would like to do so during the run-time.
 
   if ((flags_ & PbrShader::Flag::BaseColorTexture) &&
-      (materialData_->baseColorTexture != nullptr)) {
-    shader_->bindBaseColorTexture(*materialData_->baseColorTexture);
+      (materialData_->attribute<Mn::GL::Texture2D*>("baseColorTexturePtr") !=
+       nullptr)) {
```

Comment:
Huh? I thought absence of a texture now be encoded as the texture attribute not being there at all, not being null.

Reply:
Yeah, that's an oversight.  Haven't gotten to too much of the PBR-specific stuff yet